linq4j/src/test/java/org/apache/calcite/linq4j/test/ExpressionTest.java

```diff
@@ -1719,6 +1724,60 @@ public void checkBlockBuilder(boolean optimizing, String expected) {
         Expressions.toString(Expressions.constant(set)));
   }
 
+  @Test void testLargeMethod() {
```

Comment:
I believe that there is only one test in the whole of calcite that sues method-splitting.

Need a test where splitting is set to a low positive value, e.g. 3. Just to make sure that the splitter handles gracefully the case where it's impossible to split.

Is it possible to also use method-splitting in one or two of Calcite's system tests?

Reply:
I've added the test described above and also added a test to ensure method splitting does _not_ happen when methods remain under the limit.

I'm not familiar with Calcite's system tests. Would they be under the core module?